shared/middleware/helpers/redis_property_bag.rb

```diff
@@ -0,0 +1,73 @@
+# A property bag implementation backed by Redis.
+
+# Implementation notes:  The entire property bag is stored in a single
+# value as a Redis hash. This makes it efficient to fetch tne entire property
+# bag without having to make multiple requests to Redis.
+
+require 'sinatra/base'
+require 'redis'
+
+class RedisPropertyBag
+
+  # A unique prefix for Redis property bag keys.
+  PROPERTY_BAG_KEY_PREFIX = "_pbag__"
```

Comment:
Is there a particular reason/convention to the single leading underscore, double trailing underscore on this prefix?